helm-chart/binderhub/values.yaml

```diff
@@ -188,10 +188,29 @@ jupyterhub:
         admin: true
         apiToken:
   singleuser:
-    # start jupyter notebook for the server
-    cmd: jupyter-notebook
-    # use jupyterlab for the default UI
-    defaultUrl: /lab
+    # start notebook server with lab ui as default
+    # *if available*
+    cmd:
+      - python3
+      - "-c"
+      - |
+        import os
+        import sys
+
+        try:
+            import jupyterlab
+            major = int(jupyterlab.__version__.split(".", 1)[0])
+        except Exception:
+            have_lab = False
+        else:
+            have_lab = major >= 3
+
+        if have_lab and "NotebookApp.default_url" not in " ".join(sys.argv):
+            # if recent-enough lab is available, make it the default UI
+            sys.argv.insert(1, "--NotebookApp.default_url=/lab/")
+
+        # launch the notebook server
+        os.execvp("jupyter-notebook", sys.argv)
```

Comment:
Do you think this could make signal transmission (SIGTERM for example) start to fail?

Reply:
No, that's exactly what the use of `exec` here avoids. Exec *replaces* the current process, as opposed to `Popen` which launches a new process and can have signal-relay issues.

This is the Python equivalent of 

```bash
exec jupyter-notebook "$@"
```

that we use in other container entrypoints